### PR TITLE
Change to standard

### PR DESCRIPTION
**Pull request by Swapnull**

### CODE REVIEW

**Comment by serby** [comment]
Hi Martin

I don't want eslint removed, it must remain but with the standard config. If there is a reason not to do this then I would like to understand it.

/cc @benjaminparnell

**Comment by benjaminparnell** [comment]
If your just going to have eslint with the standard config, why not just use the standard tool, as it essentially is eslint with the standard config? Or am I missing something?

**Comment by serby** [comment]
How to you then add the additional `eslint-config-standard-react` and then in the future the `eslint-config-standard-*` or make project exceptions?

Also how does the IDE know what to do?

**Comment by benjaminparnell** [comment]
I guess if you wanted to add more stuff to it then you would have to go down the route of using eslint with the standard eslint plugin. My main reason of using the `standard` package was that I personally dislike having to spend time working out weird peerDependency issues whenever I try to set it up on a project. But with the clock config removed, considering the other packages are updated more regularly, that probably won't be a problem anymore.

The standard tool has these eslint packages installed https://github.com/feross/standard/blob/master/package.json#L15-L23 so I think that would probably satisfy most of our needs, but I get that we _might_ want more fine grained control. 

There are [editor plugins](https://github.com/feross/standard#are-there-text-editor-plugins) available for standard.

**Comment by serby** [comment]
Agreed about the peers. 

I'm happy to go with your advice, if the above conditions can be met. react and jsx are the most important.

Can you add to the redo readme a bit about the standard setup and the plugins needed for sublime, atom and vim.

**Comment by serby** [comment]
@benjaminparnell I've been talking to @balaclark about his experience on Sunsavers and it seem clear that in practice there is always the need to add extra rules or change the rules on a project. 

Can you confirm that with the `standard` module you can't do easily this. If this isn't the case then we should go the `eslint` way.

**Comment by benjaminparnell** [comment]
If you want full flexibility go the `eslint` route.

**Comment by balaclark** [comment]
here is what we are doing on savers: https://github.com/clocklimited/SunSavers/blob/master/app/.eslintrc

Standard + a little extra strict checking for proptypes and spacing. So far we haven't had issues.

```json
{
  "extends": ["standard", "standard-react"],
  "parser": "babel-eslint",
  "rules": {
    "jsx-quotes": [2, "prefer-double"],
    "react/forbid-prop-types": [2, {"forbid":["object", "array"]}],
    "object-curly-spacing": ["error", "always"],
    "array-bracket-spacing": ["error", "always"]
  },
  "env": {
    "es6": true,
    "node": true,
    "browser": true,
    "mocha": true
  },
  "plugins": [
    "react"
  ]
}
```

File: .eslintrc [hunk 1]
```diff
@@ -1,8 +1,12 @@
 {
-  "extends": "clock",
-
+  "extends": "standard",
   "env": {
     "es6": false
+  },
+  "globals": {
```
Comment (serby) [inline .eslintrc]: Just add the env mocha

File: generator.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
 module.exports = NodeProjectGenerator
 
 var kickoff = require('kickoff')
-  , Generator = kickoff.Generator
-  , generate = kickoff.generate
-  , extend = require('lodash.assign')
-  , slugg = require('slugg')
+var Generator = kickoff.Generator
```
Comment (serby) [inline generator.js]: These can be const. Always try and remove var when possible.

File: source/pliers.js [hunk 3]
```diff
@@ -1,16 +1,15 @@
 module.exports = tasks
 
 var glob = require('glob')
+var path = require('path')
```
Comment (serby) [inline source/pliers.js]: Switch to consts

File: source/test/runner.js [hunk 4]
```diff
@@ -1,15 +1,15 @@
 var Mocha = require('mocha')
-  , fs = require('fs')
-  , path = require('path')
-  , glob = require('glob')
-  , mocha = new Mocha()
-  , filter = process.argv[2] && new RegExp(process.argv[2])
+var fs = require('fs')
+var path = require('path')
+var glob = require('glob')
```
Comment (serby) [inline source/test/runner.js]: Switch to consts

File: test/package-json-template.test.js [hunk 5]
```diff
@@ -1,22 +1,19 @@
...
   it('should generate valid json', function () {
-
-    var tmpl = template(fs.readFileSync(__dirname + '/../templates/package.json.tpl'))
-      , mockSettings =
-          { platform: 'test'
-          , description: 'test'
-          , organization: 'test'
-          }
-      , json = tmpl(mockSettings)
+    var tmpl = template(fs.readFileSync(path.join(__dirname, '/../templates/package.json.tpl')))
```
Comment (serby) [inline test/package-json-template.test.js]: Switch to consts

**Pull request merged**